Compile Error :  iceoryx_dds/Mempool.hpp: No such file or directory

## Compile Error

**Operating system:**
Ubuntu 18.04.4 

**Compiler version:**
gcc version 7.5.0

**iceoryx version:**
iceoryx-2.0.2

**Problem Description:**

I try build as:

Build with script

```
cd iceoryx
./tools/iceoryx_build_test.sh build-all
```

Build with CMake

```
cmake -Bbuild -Hiceoryx_meta -DBUILD_ALL=ON -DCMAKE_INSTALL_PREFIX=./install
```


Both errors occurred

```
iceoryx-2.0.2/iceoryx_dds/include/iceoryx_dds/dds/cyclone_data_reader.hpp:21:10: fatal error: iceoryx_dds/Mempool.hpp: No such file or directory
```


I would like to know how to solve this problem, thanks a lot

> @quanqixian Could you post an anonymized build log of your iceoryx build? That would help to understand the issue.

This is the final terminal output when the error occurs, and [here](https://github.com/quanqixian/iceoryx_build_log/blob/main/README.md) is the full compilation output

```
[ 64%] Linking CXX executable ../iox-introspection-client
[ 64%] Built target iox-introspection-client
[ 64%] Building CXX object iceoryx_dds/CMakeFiles/iceoryx_dds.dir/source/iceoryx_dds/log/logging.cpp.o
[ 64%] Building CXX object iceoryx_dds/CMakeFiles/iceoryx_dds.dir/source/iceoryx_dds/dds/cyclone_context.cpp.o
[ 64%] Building CXX object iceoryx_dds/CMakeFiles/iceoryx_dds.dir/source/iceoryx_dds/dds/cyclone_data_reader.cpp.o
In file included from /home/quan/gitee/OpenSourceSoftware/iceoryx/iceoryx_dds/source/iceoryx_dds/dds/cyclone_data_reader.cpp:18:0:
/home/quan/gitee/OpenSourceSoftware/iceoryx/iceoryx_dds/include/iceoryx_dds/dds/cyclone_data_reader.hpp:21:10: fatal error: iceoryx_dds/Mempool.hpp: No such file or directory
 #include "iceoryx_dds/Mempool.hpp"
          ^~~~~~~~~~~~~~~~~~~~~~~~~
compilation terminated.
iceoryx_dds/CMakeFiles/iceoryx_dds.dir/build.make:103: recipe for target 'iceoryx_dds/CMakeFiles/iceoryx_dds.dir/source/iceoryx_dds/dds/cyclone_data_reader.cpp.o' failed
make[2]: *** [iceoryx_dds/CMakeFiles/iceoryx_dds.dir/source/iceoryx_dds/dds/cyclone_data_reader.cpp.o] Error 1
CMakeFiles/Makefile2:1444: recipe for target 'iceoryx_dds/CMakeFiles/iceoryx_dds.dir/all' failed
make[1]: *** [iceoryx_dds/CMakeFiles/iceoryx_dds.dir/all] Error 2
Makefile:155: recipe for target 'all' failed
make: *** [all] Error 2
quan@iZ2zeeq1jai7e6qn7xqrnuZ:~/gitee/OpenSourceSoftware/iceoryx$ 
quan@iZ2zeeq1jai7e6qn7xqrnuZ:~/gitee/OpenSourceSoftware/iceoryx$ find ./ -name Mempool.hpp
quan@iZ2zeeq1jai7e6qn7xqrnuZ:~/gitee/OpenSourceSoftware/iceoryx$ 
```

--- a/doc/website/release-notes/iceoryx-unreleased.md
+++ b/doc/website/release-notes/iceoryx-unreleased.md
@@ -19,6 +19,7 @@
 
 - FreeBSD CI build is broken [\#1338](https://github.com/eclipse-iceoryx/iceoryx/issues/1338)
 - High CPU load in blocked publisher is reduced by introducing smart busy loop waiting (adaptive_wait) [\#1347](https://github.com/eclipse-iceoryx/iceoryx/issues/1347)
+- Compile Error : iceoryx_dds/Mempool.hpp: No such file or directory [\#1364](https://github.com/eclipse-iceoryx/iceoryx/issues/1364)
 
 **Refactoring:**
 

--- a/iceoryx_dds/CMakeLists.txt
+++ b/iceoryx_dds/CMakeLists.txt
@@ -118,6 +118,8 @@ if(USE_CYCLONE_DDS)
 
     add_custom_target(mempool_header ALL DEPENDS ${CMAKE_CURRENT_BINARY_DIR}/Mempool.hpp)
 
+    add_dependencies(iceoryx_dds mempool_header)
+
     target_include_directories(iceoryx_dds PUBLIC
         $<BUILD_INTERFACE:${CMAKE_BINARY_DIR}>
         $<INSTALL_INTERFACE:include/${PREFIX}>

